### ISSUE TITLE
False positive for variable used in data provider

### ISSUE THREAD

**Issue by tb102122** [terraform-linters/tflint-ruleset-aws#446]
The rule validation is not working correctly if the variable is only used in a data provider.
check to reproduce:

```
variable "buckert_name" {
  description = "Bucket Name for Assets to be used in EC2"
  type        = string
  default     = ""
}

data "aws_s3_bucket" "selected" {
  bucket   = var.bucket_name
}
```
Result:

variables.tf:X:Y: Warning - variable "buckert_name" is declared but not used (terraform_unused_declarations)
